src/plugins/paginate/paginate.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * @title WET-BOEW Tag filter
+ * @overview Filter based content tagging
+ * @license wet-boew.github.io/wet-boew/License-en.html / wet-boew.github.io/wet-boew/Licence-fr.html
+ * @author @duboisp
+ */
+( function( $, window, document, wb ) {
+"use strict";
+
+let i18n, i18nText;
+
+const componentName = "wb-paginate",
+	selector = ".provisional." + componentName,
+	initEvent = "wb-init" + selector,
+	$document = wb.doc,
+	filterClass = "wb-fltr-out",
+	pgFilterOutClass = "wb-pgfltr-out",
+	tagFilterClass = "wb-tgfltr-out",
+	pagerClass = "wb-paginate-pager",
+	pageData = "data-pagination-idx",
+	notFilterClassSel = ":not(." + filterClass + "):not(." + tagFilterClass + ")",
+	defaults = {
+		lst: {
+			selector: "li"
+		},
+		grp: {
+			selector: "> *"
+		},
+		tbl: {
+			selector: "tr",
+			section: ":scope > tbody"
+		}
+	},
+
+	init = function( event ) {
+		const elm = wb.init( event, componentName, selector );
+
+		if ( elm ) {
+			var userConf = JSON.parse( elm.getAttribute( "data-wb-paginate" ) ),
+				paginationElm,
+				elmTagName = elm.nodeName,
+				setDefault,
+				uiTargetElm;
+
+			if ( !i18nText ) {
+				i18n = wb.i18n;
+				i18nText = {
+					prv: i18n( "prv" ),
+					nxt: i18n( "nxt" )
+				};
+			}
+
+			// Setup configurations
+			switch ( elmTagName ) {
+			case "UL":
+				setDefault = defaults.lst;
+				break;
+			case "TABLE":
+				setDefault = defaults.tbl;
+				break;
+			default:
+				setDefault = defaults.grp;
+				break;
+			}
+
+			elm.id = elm.id || wb.getId();
+			elm.pgSettings = { ...setDefault, ...userConf };
+			elm.pgSettings.currPage = 1;
+			elm.pgSettings.itemsPerPage = elm.pgSettings.itemsPerPage || 25;
+			elm.pgSettings.items = elm.querySelectorAll( ( elm.pgSettings.section || ":scope" ) + " " + elm.pgSettings.selector + notFilterClassSel );
+
+			// Setup pagination container
+			paginationElm = document.createElement( "div" );
+			paginationElm.id = componentName + "-" + elm.id;
+			paginationElm.classList.add( pagerClass );
+
+			// Add pagination container
+			if ( elm.pgSettings.uiTarget ) {
+				uiTargetElm = document.querySelector( elm.pgSettings.uiTarget );
+				uiTargetElm.appendChild( paginationElm );
+			} else if ( elm.pgSettings.section ) {
+				if ( elmTagName === "UL" || elmTagName === "TABLE" ) {
+					elm.after( paginationElm );
+				} else {
+					elm.querySelector( elm.pgSettings.section ).after( paginationElm );
+				}
+			} else {
+				elm.after( paginationElm );
+			}
+
+			// Show/hide items and generate pagination
+			updateItems( elm );
+			generateUI( elm );
+
+			wb.ready( $( elm ), componentName );
+		}
+	},
+
+	// Set or reset pagination UI
+	generateUI = function( elm ) {
+		var paginationUI = "",
+			currPage = elm.pgSettings.currPage,
+			pagesCount = elm.pgSettings.pagesCount,
```

Comment:
Comment

There is no demo for the configuration:
* currPage
* pagesCount

But I figured out it is a dynamic data managed by the plugin. I like the idea and it allow other plugin to gather the current state of the pagination plugin.